### DISCUSSION TITLE
Question : SEO , Robots

### DISCUSSION THREAD

**Discussion by leobenkel** [thebuilder/react-intersection-observer#493] [Q&A]
Hello,

I am curious as to the behavior of this in term of SEO and how google bot, for instance, would see the page.

Let's say I do something simple like:

```
{ isVisible ? <div>...</div> : null }
```

in my React component.

Will the bot be able to see the element ? Is everything considered "visible" for the bot ? 

Thank you !

**Comment by thebuilder** [comment]
As a general rule, no - It won't be seen. But Google Bot does load and execute the page, including JS. So it could in theory detect your element.

If it it actual content on your page, consider changing the approach you use to show it. It should always be added to the DOM, but hidden with `opacity: 0`. This will allow you to animate the element in to view, and also avoid page reflow/jumping content.

See triggering animations for an example:
https://github.com/thebuilder/react-intersection-observer/blob/master/docs/Recipes.md#trigger-animations

**Comment by leobenkel** [comment]
Yes, I am using the `opacity` already. I wanted to double check the details related to SEO :)
